### PR TITLE
Sia380 - Adaptive comfort setbacks

### PR DESCRIPTION
**Pull request by maxenceryan**
- Add setbacks to properties derived from SIA 2024
  - setbacks are actually NOT part of SIA 2024, so +/- 2*C implemented
  - some room types are exempted as temperatures are likely not only for people but also experiments / equipment
    - "9.3 Laborraum",                    
    - "11.2 Fitnessraum",                   
    -  "11.3 Schwimmhalle",                   
    -  "12.11 Kuehlraum",                    
    - "12.12 Serverraum"
  - passed through the room properties json into sia380
    - added fields in building input form
    - added toggle for adaptive comfort
    - added input toggle for sia380 component
  - pre process SIA 2024 schedules to standardize their input
    - python script for generating JSON updated
    - moved schedule generators mostly outside sia380
    - created new setpoint/setback schedule, which is based on occupancy and a 2 hour buffer before occupied hours
  - added to sia380
- Change how eta_g used in hourly
  - eta_g depends on a time latency of the building (tau), which doesnt make much sense / needs to be adapted for hourly calc
  - for hourly, we use the same eta_g over the month
  - this required rearrange main loop to accommodate for monthly eta_g calc for hourly demand calc
- Enable adpative comfort via a checkbox in Building Input Form UI
- Updated installer
- General Cleanup renamings, refactor

### CODE REVIEW

**Comment by christophwaibel** [comment]
When set to hourly, sia380 component doesn't work. Results in `SolutionException:DevicesSchedule` error. 
I did recompile the jsons beforehand.

image

File: src/Hive.IO/Hive.IO/Building/compile_sia2024_json.py [hunk 1]
```diff
@@ -4,14 +4,14 @@
 import csv
 import json
 
-from yaml.loader import FullLoader
-
 # map the csv header names to Sia2024Record names...
 # (try to keep them as similar as possible)
 header_map = {
```
Comment (christophwaibel) [inline src/Hive.IO/Hive.IO/Building/compile_sia2024_json.py:9]: More related to your previous PR on the installer... Could you add `compile_sia2024_json.py` and `generate_sia2024_schedules_json.py`to the `build.cmd`?

**Review by christophwaibel — CHANGES_REQUESTED** [review]
Checkbox `Adaptive Comfort` in the `GhBuilding` doesn't seem to work? No changes in Q_Heat and Q_Cool in the sia380 component:

image

image

File: src/Hive.IO/Hive.IO/Building/compile_sia2024_json.py (hunk 1 above)
Comment (maxenceryan) [reply src/Hive.IO/Hive.IO/Building/compile_sia2024_json.py:9]: i will add this to the other PR

**Comment by maxenceryan** [comment]
> Checkbox `Adaptive Comfort` in the `GhBuilding` doesn't seem to work? No changes in Q_Heat and Q_Cool in the sia380 component:
> 
hm seems to work ok for me? just pushed a recompiled installer, although i imagined this wasnt the problem. Checkbox seemed to work also.

my results also looked (slightly) different. I am using 

GrasshopperExamples\Testing\Hive_Demand_Hourly.gh
GrasshopperExamples\LectureExercises\HIVE_Zurich.epw

with
image

without 
image

**Comment by maxenceryan** [comment]
odd error with Q_s_tree_unobstructed.... 

some cryptic grasshopper breakpoint that occurs twice (one for each branch of tree presumably) when i reconnect it. 




tree seems to have an expected solar curve on both branches...

**Review by christophwaibel — APPROVED** [review]

**Comment by christophwaibel** [comment]
closes #570

**Pull request merged**